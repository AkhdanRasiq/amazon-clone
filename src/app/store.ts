import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import productReducer from '../features/productSlice'
import alertReducer from '../features/alertSlice'

export const store = configureStore({
  reducer: {
    products  : productReducer,
    alert     : alertReducer
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
