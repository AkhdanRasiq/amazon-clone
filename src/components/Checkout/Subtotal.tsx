import React, { useEffect, useState } from 'react'

import CurrencyFormat from 'react-currency-format'
import { useAppSelector } from '../../app/hooks'
import { selectProductList } from '../../features/productSlice'


function Subtotal() {
  const [totalPrice, setTotalPrice] = useState<string>()
  const [totalItem, setTotalItem]   = useState<string>()
  const productList                 = useAppSelector(selectProductList)

  useEffect(() => {
    updateSubtotal()
  }, [productList])

  const updateSubtotal = () => {
    setTotalPrice(() => {
      return productList.reduce((prev, cur) => prev + (cur.price * cur.qty), 0).toFixed(2).toString()
    })
    setTotalItem(() => {
      return productList.reduce((prev, cur) => prev + cur.qty, 0).toString()
    })
  }

  return (
    <div className="subtotal">
      <CurrencyFormat
        renderText={(value) => (
          <>
            <p>
              Subtotal ({totalItem} Items):
              <strong>{value}</strong>
            </p>
            <small className='subtotalGift'>
              <input type="checkbox" /> This order contains a gift
            </small>
          </>
        )}
        decimalScale={2}
        value={totalPrice}
        displayType={"text"}
        thousandSeparator={true}
        prefix={"$"}
      />
      <button >Proceed to Checkout</button>
    </div>
  )
}

export default Subtotal
