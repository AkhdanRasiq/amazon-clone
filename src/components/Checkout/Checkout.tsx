import React from "react"
import Subtotal from "./Subtotal"

function Checkout() {
  return (
    <div className="checkout">
      <div className="checkoutItems">
        <div>
          <h2 className="checkoutTitle">
            Your Shopping Basket
          </h2>

        </div>
      </div>
      <div className="checkoutSubtotal">
        <Subtotal />
      </div>
    </div>
  )
}

export default Checkout
