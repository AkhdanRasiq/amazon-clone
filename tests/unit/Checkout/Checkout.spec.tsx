import React from 'react'
import { render } from "@testing-library/react"
import { store } from '../../../src/app/store'
import { Provider } from 'react-redux'

import Checkout from "../../../src/components/Checkout/Checkout"

describe("Checkout.tsx", () => {
  it("Render Properly", async () => {
    const wrapper = render(
      <Provider store={store}>
        <Checkout />
      </Provider>
    )
    await expect(wrapper).toMatchSnapshot()
  })
})
