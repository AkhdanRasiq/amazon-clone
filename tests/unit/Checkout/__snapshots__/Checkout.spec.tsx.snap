// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout.tsx Render Properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="checkout"
      >
        <div
          class="checkoutItems"
        >
          <div>
            <h2
              class="checkoutTitle"
            >
              Your Shopping Basket
            </h2>
            <div
              class="productCheckout"
            >
              <div>
                <img
                  alt="product_img"
                  class="productCheckoutImage"
                  src="www.test.com"
                />
              </div>
              <div
                class="productCheckoutInfo"
              >
                <p
                  class="productCheckoutInfoTitle"
                >
                  Test
                </p>
                <p
                  class="productCheckoutInfoPrice"
                >
                  <small>
                    $
                  </small>
                  <strong>
                    19.99
                  </strong>
                </p>
                <div
                  class="productCheckoutInfoRating"
                >
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                </div>
                <div
                  class="productCheckoutAction"
                >
                  <button>
                    ─
                  </button>
                  <p>
                    3
                  </p>
                  <button>
                    +
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="checkoutSubtotal"
        >
          <div
            class="subtotal"
          >
            <p
              id="detailTotal"
            >
              Subtotal (
              3
               Items):
              <strong>
                $59.97
              </strong>
            </p>
            <small
              class="subtotalGift"
            >
              <input
                type="checkbox"
              />
               This order contains a gift
            </small>
            <button>
              Proceed to Checkout
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="checkout"
    >
      <div
        class="checkoutItems"
      >
        <div>
          <h2
            class="checkoutTitle"
          >
            Your Shopping Basket
          </h2>
          <div
            class="productCheckout"
          >
            <div>
              <img
                alt="product_img"
                class="productCheckoutImage"
                src="www.test.com"
              />
            </div>
            <div
              class="productCheckoutInfo"
            >
              <p
                class="productCheckoutInfoTitle"
              >
                Test
              </p>
              <p
                class="productCheckoutInfoPrice"
              >
                <small>
                  $
                </small>
                <strong>
                  19.99
                </strong>
              </p>
              <div
                class="productCheckoutInfoRating"
              >
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
              </div>
              <div
                class="productCheckoutAction"
              >
                <button>
                  ─
                </button>
                <p>
                  3
                </p>
                <button>
                  +
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="checkoutSubtotal"
      >
        <div
          class="subtotal"
        >
          <p
            id="detailTotal"
          >
            Subtotal (
            3
             Items):
            <strong>
              $59.97
            </strong>
          </p>
          <small
            class="subtotalGift"
          >
            <input
              type="checkbox"
            />
             This order contains a gift
          </small>
          <button>
            Proceed to Checkout
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
